sympy/integrals/risch.py

```diff
@@ -1283,7 +1283,10 @@ def integrate_primitive_polynomial(p, DE):
             aa, ad = frac_in(a, DE.t)
 
             try:
-                (ba, bd), c = limited_integrate(aa, ad, [(Dta, Dtb)], DE)
+                rv = limited_integrate(aa, ad, [(Dta, Dtb)], DE)
+                if rv is None:
+                    raise NonElementaryIntegralException
+                (ba, bd), c = rv
                 if len(c) != 1:
```

Comment:
Oh, of course. As for this case `g1, g2`, will be there => `c1, c2` with c1 = 1 (fixed) we get v = [c2] hence a fixed length.